src/Symfony/Component/Messenger/Transport/Serialization/Serializer.php

```diff
@@ -82,6 +84,10 @@ public function decode(array $encodedEnvelope): Envelope
         try {
             $message = $this->serializer->deserialize($encodedEnvelope['body'], $encodedEnvelope['headers']['type'], $this->format, $context);
         } catch (UnexpectedValueException $e) {
+            if (!class_exists(PropertyAccessor::class) && false !== strpos($e->getMessage(), 'no supporting normalizer found')) {
+                throw new LogicException('No denormalizer found to decode message. Try running "composer require symfony/property-access".', $e->getCode(), $e);
```

Comment:
Should it still throw a `MessageDecodingFailedException`?

Also... checking the exception message is not ideal (as you know). Perhaps we could make this *slightly* less smart, but still useful by:

1) Only check for the `class_exists()` (not the string stuff)
2) Use the same exception and exception message as below, but with an added sentence - e.g.

> Could not decode message: %s. Try running "composer require symfony/property-access".

Also, question: what exact situation does this cover? I'm just not familiar. If the PropertyAccessor component is missing, does it sometimes fail to deserialize the message? I know that, in some situations, it can cause an object to be deserialized by with some missing data. But what's the situation here? Is it if you have a `__construct()` function but are missing the `PropertyAccessor`?

Reply:
I agree with your point against checking the exception message. We can skip the error check.

This exception is thrown when no (de)normalizer is found that can handle objects. When `property-access` is available, the `ObjectNormalizer` of `symfony/serializer` is available. Without it, though, it is deregistered from the container and no longer available.

However, in the rare case that someone has another (3rd party or their own) (de)normalizer that can denormalize objects, this exception could be thrown when that fails (for whatever reason), even though a denormalizer is available.

In a perfect world, I would add a few extra Exceptions to the serializer to distinguish between these scenarios (a difference between "no (de)normalizer supports this" and "something went wrong while using the (de)normalizer we found".